src/app/detail/ui/TaskEditor.tsx

```diff
@@ -152,23 +154,28 @@ export const TaskEditor = ({
 
       <Box mt="12px" sx={{ height: '100%' }}>
         <Tapwrite
-          uploadFn={async (file, tiptapEditorUtils) => {
-            const newBlob = await upload(file.name, file, {
-              access: 'public',
-              handleUploadUrl: '/api/upload',
-            })
-            tiptapEditorUtils.setImage(newBlob.url as string)
-          }}
           content={updateDetail}
           getContent={handleDetailChange}
           readonly={userType === UserType.CLIENT_USER}
           editorClass="tapwrite-details-page"
           placeholder="Add description..."
+          uploadFn={async (file, tiptapEditorUtils) => {
+            const supabaseActions = new SupabaseActions()
+            const fileName = generateRandomString(file.name)
+            const signedUrl: ISignedUrlUpload = await getSignedUrlUpload(fileName)
+            const filePayload = await supabaseActions.uploadAttachment(file, signedUrl, task_id)
+            const url = await supabaseActions.getPublicUrl(filePayload?.filePath ?? '')
+            tiptapEditorUtils.setImage(url, fileName)
+          }}
+          deleteEditorAttachments={async (id: string) => {
```

Comment:
Can you please explain what is this `id`? It is the `filename` or something else and how is it generated? 
I am asking this because the `Tapwrite` is open sourced and we shouldn't limit `id` to supabase uploads.